File: queue-frontend/src/app/dtos/attendance/statistics/ResponseAttendanceDashboardDto.ts
import { ResponseAttendancesByCustomerStatisticsDto } from "./ResponseAttendancesByCustomerStatisticsDto";
import { ResponseAttendancesByDepartmentStatisticsDto } from "./ResponseAttendancesByDepartmentStatisticsDto";
import { ResponseAttendancesByHourStatisticsDto } from "./ResponseAttendancesByHourStatisticsDto";
import { ResponseAttendancesByServiceStatisticsDto } from "./ResponseAttendancesByServiceStatisticsDto";
import { ResponseAttendancesByWeekStatisticsDto } from "./ResponseAttendancesByWeekStatisticsDto";
import { ResponseAttendancesCreatedByMonthStatisticsDto } from "./ResponseAttendancesCreatedByMonthStatisticsDto";
import { ResponseAverageAttendanceByUserStatisticsDto } from "./ResponseAverageAttendanceByUserStatisticsDto";
import { ResponseAverageServiceTimeStatisticsDto } from "./ResponseAverageServiceTimeStatisticsDto";
import { ResponseAverageWaitingTimeStatisticsDto } from "./ResponseAverageWaitingTimeStatisticsDto";
import { ResponseCountTotalAttendancesStatisticsDto } from "./ResponseCountTotalAttendancesStatisticsDto";

export interface ResponseAttendanceDashboardDto {

    countTotalAttendances: ResponseCountTotalAttendancesStatisticsDto;
    averageWaitingTime: ResponseAverageWaitingTimeStatisticsDto;
    averageServiceTime: ResponseAverageServiceTimeStatisticsDto;
    averageAttendanceByUser: ResponseAverageAttendanceByUserStatisticsDto[];
    attendancesCreatedByMonth: ResponseAttendancesCreatedByMonthStatisticsDto[];
    attendancesByWeek: ResponseAttendancesByWeekStatisticsDto[];
    attendancesByService: ResponseAttendancesByServiceStatisticsDto[];
    attendancesByHour: ResponseAttendancesByHourStatisticsDto[];
    attendancesByDepartment: ResponseAttendancesByDepartmentStatisticsDto[];
    attendancesByCustomer: ResponseAttendancesByCustomerStatisticsDto[];
}

File: queue/src/main/java/br/com/queue/dtos/attendance/statistics/ResponseAttendanceDashboardDto.java
package br.com.queue.dtos.attendance.statistics;

import java.util.List;

public record ResponseAttendanceDashboardDto(

        ResponseCountTotalAttendancesStatisticsDto countTotalAttendances,

        ResponseAverageWaitingTimeStatisticsDto averageWaitingTime,

        ResponseAverageServiceTimeStatisticsDto averageServiceTime,

        List<ResponseAverageAttendanceByUserStatisticsDto> averageAttendanceByUser,

        List<ResponseAttendancesCreatedByMonthStatisticsDto> attendancesCreatedByMonth,

        List<ResponseAttendancesByWeekStatisticsDto> attendancesByWeek,

        List<ResponseAttendancesByServiceStatisticsDto> attendancesByService,

        List<ResponseAttendancesByHourStatisticsDto> attendancesByHour,

        List<ResponseAttendancesByDepartmentStatisticsDto> attendancesByDepartment,

        List<ResponseAttendancesByCustomerStatisticsDto> attendancesByCustomer

) {
}
